UnitsNet/CustomCode/UnitAbbreviationsCache.cs

```diff
@@ -226,9 +280,13 @@ public string GetDefaultAbbreviation(UnitKey unitKey, IFormatProvider? formatPro
         /// <param name="unit">Enum value for unit.</param>
         /// <param name="formatProvider">The format provider to use for lookup. Defaults to <see cref="CultureInfo.CurrentCulture" /> if null.</param>
         /// <returns>Unit abbreviations associated with unit.</returns>
+        /// <exception cref="UnitNotFoundException">
+        ///     Thrown when no unit information is found for the specified
+        ///     <paramref name="unit" />.
+        /// </exception>
         public string[] GetUnitAbbreviations<TUnitType>(TUnitType unit, IFormatProvider? formatProvider = null) where TUnitType : struct, Enum
         {
-            return GetUnitAbbreviations(UnitKey.ForUnit(unit), formatProvider).ToArray();  // TODO can we change this to return an IReadonlyCollection (as the GetAbbreviations)?
+            return GetUnitAbbreviations(UnitKey.ForUnit(unit), formatProvider).ToArray(); // TODO can we change this to return an IReadOnlyList (as the GetAbbreviations)?
```

Comment:
Let's do it, we are more free to change the underlying type later

Reply:
👍 After changing this to `IReadOnlyList` - there shouldn't be any need for either `ToArray` or `ToList` .

Reply:
fixed in https://github.com/angularsen/UnitsNet/pull/1548/commits/f9a2766b959334432c531c44afca207f37c155cc